pkg/metadata/metadata.go

```diff
@@ -46,158 +43,176 @@ import (
 	ipnimd "github.com/ipni/go-libipni/metadata"
 	"github.com/multiformats/go-multicodec"
 	"github.com/multiformats/go-varint"
-	"github.com/storacha-network/indexing-service/pkg/capability/assert"
 )
 
 var (
-	_ ipnimd.Protocol = (*ContentClaimMetadata)(nil)
+	_ ipnimd.Protocol = (*IndexClaimMetadata)(nil)
 
 	//go:embed metadata.ipldsch
-	schemaBytes          []byte
-	contentClaimMetadata schema.TypedPrototype
+	schemaBytes                []byte
+	indexClaimMetadata         schema.TypedPrototype
+	equalsClaimMetadata        schema.TypedPrototype
+	locationCommitmentMetadata schema.TypedPrototype
 )
 
 func init() {
 	typeSystem, err := ipld.LoadSchemaBytes(schemaBytes)
 	if err != nil {
 		panic(fmt.Errorf("failed to load schema: %w", err))
 	}
-	t := typeSystem.TypeByName("ContentClaimMetadata")
-	contentClaimMetadata = bindnode.Prototype((*ContentClaimMetadata)(nil), t)
+	indexClaimMetadata = bindnode.Prototype((*IndexClaimMetadata)(nil), typeSystem.TypeByName("IndexClaimMetadata"))
+	equalsClaimMetadata = bindnode.Prototype((*EqualsClaimMetadata)(nil), typeSystem.TypeByName("EqualsClaimMetadata"))
+	locationCommitmentMetadata = bindnode.Prototype((*LocationCommitmentMetadata)(nil), typeSystem.TypeByName("LocationCommitmentMetdata"))
 }
 
+// metadata identifiers
 // currently we just use experimental codecs for now
-const TransportContentClaim = 0x3E0000
 
-type ClaimType uint64
+// IndexClaimID is the multicodec for index claims
+const IndexClaimID = 0x3E0000
 
-const (
-	LocationCommitment ClaimType = iota
-	IndexClaim
-	EqualsClaim
-)
+// EqualsClaimID is the multicodec for equals claims
+const EqualsClaimID = 0x3E0001
+
+// LocationCommitmentID is the multicodec for location commitments
+const LocationCommitmentID = 0x3E0002
+
+var nodePrototypes = map[multicodec.Code]schema.TypedPrototype{
+	IndexClaimID:         indexClaimMetadata,
+	EqualsClaimID:        equalsClaimMetadata,
+	LocationCommitmentID: locationCommitmentMetadata,
+}
+
+var MetadataContext ipnimd.MetadataContext
+
+func init() {
+	mdctx := ipnimd.Default
+	mdctx = mdctx.WithProtocol(IndexClaimID, func() ipnimd.Protocol { return &IndexClaimMetadata{} })
+	mdctx = mdctx.WithProtocol(EqualsClaimID, func() ipnimd.Protocol { return &EqualsClaimMetadata{} })
+	mdctx = mdctx.WithProtocol(LocationCommitmentID, func() ipnimd.Protocol { return &LocationCommitmentMetadata{} })
+	MetadataContext = mdctx
+}
+
+type HasClaimCid interface {
+	GetClaimCid() cid.Cid
```

Comment:
let me tell you my thoughts some time on the totally useless and unneccesary abstraction that is ipld.Link. The problem is there is no other implementation other than cidlink. It exists cause the go-ipld-prime author felt the cid library was too heavyweight. Anyway, I'm not gonna fight gravity we can just use ipld.Link if you feel strongly but I think it just creates weird code with a lot of link.(cidlink.Link).Cid casts, which mind you might cause panics if there ever is another link type implemented. Personally, one of the great parts of bindnode is it gets you out of having to use ipld.Link's everywhere since it will translate link schemas to cid.Cid.

Reply:
I guess my preference would be to use it...I can relate to your feelings towards it, but I think using the majority of ipld prime but rejecting this one part makes the code we write look like it's half old and half new and a bit esoteric.

A few other reasons:

* JS and rust IPLD libraries have also embraced `Link` and a lot of our existing JS code uses `Link`.
* In IPLD schemas it is a `Link`, so might be confusing to find `cid.Cid` in go code?
* I don't remember doing a _lot_ of `link.(cidlink.Link).Cid` in all the go code I've written using IPLD, although granted, I have _sometimes_ had to do something similar. I think mostly for working with older code that uses `CID` not `Link` though?